core/trino-spi/src/main/java/io/trino/spi/block/AbstractRowBlock.java

```diff
@@ -65,6 +65,20 @@ protected AbstractRowBlock(int numFields)
         this.numFields = numFields;
     }
 
+    public void writeRowsTo(int[] positions, int offset, int length, BatchRowWriter batchWriter)
+    {
+        int[] fieldPositions = new int[length];
+        for (int i = offset; i < offset + length; i++) {
+            int position = positions[i];
+            if (isNull(position)) {
+                throw new IllegalArgumentException("Position is null " + position);
+            }
+            fieldPositions[i] = getFieldBlockOffset(position);
+        }
+
+        batchWriter.writeRows(getRawFieldBlocks(), fieldPositions, length);
```

Comment:
Could you do a rough check how bad the overhead would be in your JMH benchmarks ? If it's only upto 10% worse then it might be better to stick to existing abstractions (unless the new abstraction is planned to have more use cases in future changes).

Reply:
this depends on the case. For a Row with `BIGINT` it's between 20% and 90%
```
PR
Benchmark                                   (channelCount)  (enableCompression)  (nullRate)  (partitionCount)  (positionCount)             (type)  Mode  Cnt    Score    Error  Units
BenchmarkPartitionedOutputOperator.addPage               1                false           0                16             8192  ROW_BIGINT_BIGINT  avgt   10  152.278 ±  1.341  ms/op
BenchmarkPartitionedOutputOperator.addPage               1                false         0.2                16             8192  ROW_BIGINT_BIGINT  avgt   10  262.213 ±  4.140  ms/op
BenchmarkPartitionedOutputOperator.addPage               2                false           0                16             8192  ROW_BIGINT_BIGINT  avgt   10  227.130 ±  2.520  ms/op
BenchmarkPartitionedOutputOperator.addPage               2                false         0.2                16             8192  ROW_BIGINT_BIGINT  avgt   10  516.319 ± 12.261  ms/op
BenchmarkPartitionedOutputOperator.addPage               4                false           0                16             8192  ROW_BIGINT_BIGINT  avgt   10  396.421 ±  8.581  ms/op
BenchmarkPartitionedOutputOperator.addPage               4                false         0.2                16             8192  ROW_BIGINT_BIGINT  avgt   10  900.062 ±  9.274  ms/op

getPositions
Benchmark                                   (channelCount)  (enableCompression)  (nullRate)  (partitionCount)  (positionCount)             (type)  Mode  Cnt     Score    Error  Units
BenchmarkPartitionedOutputOperator.addPage               1                false           0                16             8192  ROW_BIGINT_BIGINT  avgt   10   195.421 ±  1.665  ms/op
BenchmarkPartitionedOutputOperator.addPage               1                false         0.2                16             8192  ROW_BIGINT_BIGINT  avgt   10   340.247 ± 10.956  ms/op
BenchmarkPartitionedOutputOperator.addPage               2                false           0                16             8192  ROW_BIGINT_BIGINT  avgt   10   442.651 ± 12.785  ms/op
BenchmarkPartitionedOutputOperator.addPage               2                false         0.2                16             8192  ROW_BIGINT_BIGINT  avgt   10   601.633 ± 31.613  ms/op
BenchmarkPartitionedOutputOperator.addPage               4                false           0                16             8192  ROW_BIGINT_BIGINT  avgt   10   759.232 ± 13.089  ms/op
BenchmarkPartitionedOutputOperator.addPage               4                false         0.2                16             8192  ROW_BIGINT_BIGINT  avgt   10  1257.812 ± 12.178  ms/op
```